### ISSUE TITLE
Can't recreate Iso et al.'s results

### ISSUE THREAD

**Issue by jqhoogland** [jqhoogland/mlrg#1]
Some ideas:

Something going wrong with the sampling?
How to check: build a thermometer also measure magnetization

Overfitting? Try with regularization.
Also, figure out a way to plot both training set and testing set error during training. Perhaps 5000 epochs is too much. Also, how does the batch size work? I didn't fully understand this part of the RBM. Did Iso et al. calculate ensemble averages over the entire test set each time? Or did they do SGD?

Do Iso et al. "measure" the spins after each gibbs step? i.e. do they leave the mean field approximation without probabilistically firing the spins- if they don't, this would leave many more accessible/spurious minima over. Could they be suppressed by the randomness (temperature eliminates spurious states in hopfield model). Indeed, it seems that mf chains converge to a handful of possible states.
